src/event-manager/event-channel-client.ts

```diff
@@ -85,40 +85,44 @@ export default class EventChannelClient {
       }
 
       this._endpointUrl = url;
-      this._wsClient = new WebSocket(url, [], { handshakeTimeout: connectionOption.handshakeTimeout || DEFAULT_HANDSHAKE_TIMEOUT_MS });
-      this._wsClient.on('message', (message: string) => {
-        this.handleMessage(message);
-      });
-      this._wsClient.on('error', (err) => {
-        console.error(err);
+      this._ws = new WebSocket(url, [], { handshakeTimeout: connectionOption.handshakeTimeout || DEFAULT_HANDSHAKE_TIMEOUT_MS });
+      this._ws.onmessage = (event: { data: unknown }) => {
+        if (typeof event.data !== 'string') {
+          return;
+        }
+        this.handleMessage(event.data);
+      };
+      this._ws.onerror = async (event: unknown) => {
+        console.error(event);
         this.disconnect();
-      });
-      this._wsClient.on('open', () => {
+      };
+      this._ws.onopen = () => {
         this._isConnected = true;
         this.startHeartbeatTimer(connectionOption.heartbeatIntervalMs || DEFAULT_HEARTBEAT_INTERVAL_MS);
         resolve(this);
-      });
-      this._wsClient.on('ping', () => {
-        if (this._heartbeatTimeout) {
-          clearTimeout(this._heartbeatTimeout);
-        }
-        this.startHeartbeatTimer(connectionOption.heartbeatIntervalMs || DEFAULT_HEARTBEAT_INTERVAL_MS);
-      });
-      this._wsClient.on('close', () => {
+      };
+      // NOTE(jiyoung): ping is not supported in browser api
```

Comment:
Isn't the goal to support both backend side and browser side? @jiyoung-an

Reply:
maybe browser-or-node package can be used?
https://stackoverflow.com/questions/34550890/how-to-detect-if-script-is-running-in-browser-or-in-node-js

Reply:
In isomorphic-ws, there is no separate ping method, so it seems that a separate message must be sent.
ref) https://github.com/fugle-dev/fugle-marketdata-node/blob/7501aa634f625ec19a101d30c7072cd523439649/src/websocket/client.ts#L45

Reply:
@platfowner I found reference code with handshake timeout and ping
ref) https://github.com/shapeshift/web/blob/4fd287342223d984f4d499ad171a62297136ec8d/packages/unchained-client/src/websocket.ts#L2

Reply:
great, thanks! I'll take a look very soon.